File: frontend/types/analytics.ts
export interface GenreStats {
  genre: string;
  total_books: number;
  total_copies: number;
  available_copies: number;
  percentage: number;
}

export interface MonthlyLendingStats {
  month: string;
  total_loans: number;
  returned_loans: number;
  overdue_loans: number;
  active_loans: number;
}

export interface TopBookStats {
  id: string;
  title: string;
  author: string;
  genre: string | null;
  shelf_location: string | null;
  copies_total: number;
  copies_available: number;
  total_borrows: number;
  current_borrows: number;
  utilization_rate: number;
}

export interface SummaryStats {
  total_books: number;
  total_members: number;
  active_loans: number;
  overdue_loans: number;
  total_copies: number;
  available_copies: number;
  loans_today: number;
  returns_today: number;
}

File: backend/app/schemas/analytics.py
from typing import Optional
from uuid import UUID

from pydantic import BaseModel


class GenreStats(BaseModel):
    genre: str
    total_books: int
    total_copies: int
    available_copies: int
    percentage: float


class MonthlyLendingStats(BaseModel):
    month: str
    total_loans: int
    returned_loans: int
    overdue_loans: int
    active_loans: int


class TopBookStats(BaseModel):
    id: UUID
    title: str
    author: str
    genre: Optional[str] = None
    shelf_location: Optional[str] = None
    copies_total: int
    copies_available: int
    total_borrows: int
    current_borrows: int
    utilization_rate: float


class SummaryStatsResponse(BaseModel):
    total_books: int
    total_members: int
    active_loans: int
    overdue_loans: int
    total_copies: int
    available_copies: int
    loans_today: int
    returns_today: int
